pydantic_ai_slim/pydantic_ai/models/google.py

```diff
@@ -181,6 +183,18 @@ async def request(
         response = await self._generate_content(messages, False, model_settings, model_request_parameters)
         return self._process_response(response)
 
+    async def count_tokens(
+        self,
+        messages: list[ModelMessage],
+    ) -> BaseCountTokensResponse:
+        check_allow_model_requests()
+        _, contents = await self._map_messages(messages)
+        response = self.client.models.count_tokens(
+            model=self._model_name,
+            contents=contents,
```

Comment:
We should include not just the messages but the entire `generateContentRequest` as function definitions etc also count as tokens